docs/reference/quickstart/run-elasticsearch-locally.asciidoc

```diff
@@ -0,0 +1,177 @@
+[[run-elasticsearch-locally]]
+== Run {es} locally in Docker (without security)
+++++
+<titleabbrev>Local dev setup (Docker)</titleabbrev>
+++++
+
+[WARNING]
+====
+*DO NOT USE THESE INSTRUCTIONS FOR PRODUCTION DEPLOYMENTS*
+
+The instructions on this page are for *local development only*. Do not use these instructions for production deployments, because they are not secure.
+While this approach is convenient for experimenting and learning, you should never run the service in this way in a production environment.
+
+Refer to https://www.elastic.co/guide/en/elasticsearch/reference/current/install-elasticsearch.html[Install {es}] to learn about the various options for installing {es} in a production environment, including using Docker.
+====
+
+The following commands help you very quickly spin up a single-node {es} cluster, together with {kib} in Docker.
+Note that if you don't need the {kib} UI, you can skip those instructions.
+
+[discrete]
+[[local-dev-why]]
+=== When would I use this setup?
+
+Use this setup if you want to quickly spin up {es} (and {kib}) for local development or testing.
+
+For example you might:
+
+* Want to run a quick test to see how a feature works.
+* Follow a tutorial or guide that requires an {es} cluster, like our <<getting-started,quick start guide>>.
+* Experiment with the {es} APIs using different tools, like the Dev Tools Console, cURL, or an Elastic programming language client.
+* Quickly spin up an {es} cluster to test an executable https://github.com/elastic/elasticsearch-labs/tree/main/notebooks#readme[Python notebook] locally.
+
+[discrete]
+[[local-dev-prerequisites]]
+=== Prerequisites
+
+If you don't have Docker installed, https://www.docker.com/products/docker-desktop[download and install Docker Desktop] for your operating system.
+
+[discrete]
+[[local-dev-env-vars]]
+=== Set environment variables
+
+Configure the following environment variables.
+
+[source,sh]
+----
+export ELASTIC_PASSWORD="<ES_PASSWORD>"  # password for "elastic" username
+export KIBANA_PASSWORD="<KIB_PASSWORD>"   # Used _internally_ by Kibana, must be at least 6 characters long
+----
+
+[discrete]
+[[local-dev-create-docker-network]]
+=== Create a Docker network
+
+To run both {es} and {kib}, you'll need to create a Docker network:
+
+[source,sh]
+----
+docker network create elastic-net
+----
+
+[discrete]
+[[local-dev-run-es]]
+=== Run {es}
+
+Start the {es} container with the following command:
+
+ifeval::["{release-state}"=="unreleased"]
+WARNING: Version {version} has not yet been released.
+No Docker image is currently available for {es} {version}.
+endif::[]
+
+[source,sh,subs="attributes"]
+----
+docker run -p 127.0.0.1:9200:9200 -d --name elasticsearch --network elastic-net \
+  -e ELASTIC_PASSWORD=$ELASTIC_PASSWORD \
+  -e "discovery.type=single-node" \
+  -e "xpack.security.http.ssl.enabled=false" \
```

Comment:
🥰

Reply:
I love this change. Having SSL out of the way is a big deal. One question though: why didn't we go for full `xpack.security.enabled=false` for local testing setup without password?

Reply:
This was a concesion we've made.

But I personally think this is a good set up. Every other database I've used (except sqlite!) uses this style of dev solution, where there is no TLS but you still have to authenticate. For someone developing an app that will eventually run in production I think it is good to have access to the auth system, since you will obviously need to have it during development to have your app ready for prod.

Reply:
Not that this is a realistic change to happen but I feel like `discovery.type=single-node` should imply `xpack.security.http.ssl.enabled=false`. You want TLS in a distributed system but it's less important for a local installation (and why it's so much easier for PostgreSQL,... to just have a generated password and be done with it)

Reply:
> Not that this is a realistic change to happen but I feel like discovery.type=single-node should imply xpack.security.http.ssl.enabled=false.

true